sentry/src/main/java/io/sentry/JsonReflectionObjectSerializer.java

```diff
@@ -37,11 +37,7 @@ public final class JsonReflectionObjectSerializer {
       return ((Byte) object).intValue();
     } else if (object instanceof Short) {
       return ((Short) object).intValue();
-    } else if (object instanceof Integer) {
-      return object;
-    } else if (object instanceof Float) {
-      return object;
-    } else if (object instanceof Double) {
+    } else if (object instanceof Number) {
```

Comment:
I'm a bit confused why do we have special handling for byte and short and don't for others?

Reply:
There's a test that fails if the special handling is removed. I've added another test that serializes then deserializes the output again. It seems to work without the special handling. But I'm not sure if there's a reason it's there.

Reply:
Please check with @denrase

Reply:
I guess i had the wrong expectation in the test, therefore the cast to `int`. So this part should be correct now.

The intent here was to return the string, characters cast to string and all Java primitives (of which i forgot `Long` apparently 🤦), as those are pretty much the leaves of the tree we are trying to traverse.

 Number is a superclass to more than those primitives, but i'm not sure what the impact would be if we for example encounter something like `DoubleAccumulator `? 

https://docs.oracle.com/javase/8/docs/api/java/lang/Number.html

Maybe we should introduce a method `Bool isPrimitive(Object object)` (and also check for `Long`) to keep the original intent and not have to think about those other `Number` subclasses.

Reply:
Just tested `DoubleAccumulator`, `BigInteger` and `BigDecimal` locally. All of them worked. Didn't test all `Number` subtypes but I'd say risk is pretty low here.